src/python/WMCore/WMSpec/Steps/Fetchers/PileupFetcher.py

```diff
@@ -70,15 +74,21 @@ def _queryDbsAndGetPileupConfig(self, stepHelper, dbsReader, fakeSites):
             # each dataset input can generally be a list, iterate over dataset names
             blockDict = {}
```

Comment:
It divides by types so each type will have the groups of blocks but if any one type (MC or data) has multiple datasets blocks are mixed together with different dataset. I am not sure that would be the problem or Pileup is restricted only one dataset per type.